observability/micrometer-api/src/main/java/org/wildfly/extension/micrometer/api/MicrometerCdiExtension.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * Copyright 2021 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.extension.micrometer.api;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.WeakHashMap;
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.spi.AfterBeanDiscovery;
+import jakarta.enterprise.inject.spi.BeanManager;
+import jakarta.enterprise.inject.spi.BeforeShutdown;
+import jakarta.enterprise.inject.spi.Extension;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import org.wildfly.security.manager.WildFlySecurityManager;
+
+public class MicrometerCdiExtension implements Extension {
+    private static final Map<ClassLoader, MeterRegistry> REGISTRY_INSTANCES = Collections.synchronizedMap(new WeakHashMap<>());
+
+    public static MeterRegistry registerApplicationRegistry(ClassLoader classLoader, MeterRegistry registry) {
```

Comment:
This being public concerns me. I'm not sure what to do about it but it seems solvable. AIUI this module exists to expose this class and io.micrometer to the application. I bet we can handle that via some module.xml stuff without having to have the class in a different package.

Reply:
Just a +1 on making this non-public especially since it's using a static map.

Reply:
Yes, we need to fix this between Beta and Final.